packages/machine/test/integration/setup-then-set-state.spec.ts

```diff
@@ -0,0 +1,204 @@
+// gas needed, so we hard-code this number to ensure the tx completes
+const CREATE_PROXY_AND_SETUP_GAS = 6e9;
+
+// Similarly, the SetupCommitment is a `delegatecall`, so we estimate
+const SETUP_COMMITMENT_GAS = 6e9;
+
+// The AppRegistry.setState call _could_ be estimated but we haven't
+// written this test to do that yet
+const SETSTATE_COMMITMENT_GAS = 6e9;
+
+let networkId: number;
+let provider: JsonRpcProvider;
+let wallet: Wallet;
+
+// TODO: When we add a second use case of this custom mathcer,
+//       move it and its typigns into somewhere re-usable
+declare global {
+  namespace jest {
+    interface Matchers<R> {
+      toBeEq(expected: BigNumber): BigNumber;
+    }
+  }
+}
+expect.extend({
+  toBeEq(received: BigNumber, argument: BigNumber) {
+    return {
+      pass: received.eq(argument),
+      message: () => `expected ${received} not to be equal to ${argument}`
+    };
+  }
+});
+
+// TODO: This will be re-used for all integration tests, so
+//       move it somewhere re-usable when we add a new test
+beforeAll(async () => {
+  require("dotenv-safe").config();
+
+  // Can use ! because dotenv-safe ensures value is set
+  const host = process.env.DEV_GANACHE_HOST!;
+  const port = process.env.DEV_GANACHE_PORT!;
+  const mnemonic = process.env.DEV_GANACHE_MNEMONIC!;
+
+  provider = new JsonRpcProvider(`http://${host}:${port}`);
+  wallet = Wallet.fromMnemonic(mnemonic).connect(provider);
+  networkId = (await provider.getNetwork()).chainId;
+});
+
+/**
+ * This test suite tests submitting a generated Set State Commitment
+ * to the blockchain and observing the result
+ */
+describe("Scenario - open channel, update free balance, on-chain resolution", () => {
+  jest.setTimeout(JEST_TEST_WAIT_TIME);
+
+  it("should distribute resolution returned from ETH Free Balance App", async done => {
+    const relevantArtifacts = [
+      AppRegistry,
+      ETHBucket,
+      NonceRegistry,
+      StateChannelTransaction
+    ];
+
+    const network = {
+      // Fetches the values from build artifacts of the contracts needed
+      // for this test and sets the ones we don't care about to 0x0
+      MultiSend: AddressZero,
+      ETHBalanceRefund: AddressZero,
+      ...relevantArtifacts.reduce(
+        (accumulator, artifact) => ({
+          ...accumulator,
+          [artifact.contractName]: artifact.networks[networkId].address
+        }),
+        {}
+      )
+    } as NetworkContext;
+
+    const signingKeys = [
+      new SigningKey(hexlify(randomBytes(32))),
+      new SigningKey(hexlify(randomBytes(32)))
+    ].sort((a, b) =>
+      parseInt(a.address, 16) < parseInt(b.address, 16) ? -1 : 1
+    );
+
+    const users = signingKeys.map(x => x.address);
+
+    const proxyFactory = new Contract(
+      ProxyFactory.networks[networkId].address,
+      ProxyFactory.abi,
+      wallet
+    );
+
+    proxyFactory.on("ProxyCreation", async proxy => {
+      let stateChannel = new StateChannel(proxy, users).setupChannel(network);
```

Comment:
Why is `setupChannel` not a static method that'd return a new StateChannel instance?

Giving `users` to such a function more strongly parallels how the multisig contract gets setup as well.

Reply:
Because it requires a protocol execution to go from "not set up" to "set up" and I think these are two distinct stages in the lifecycle of a `StateChannel`